account-integrations/sdk/src/accounts/SafeECDSAAccontWrapper.ts

```diff
@@ -0,0 +1,249 @@
+import { ethers } from 'ethers';
+import { z } from 'zod';
+import {
+  SafeECDSAPlugin,
+  SafeECDSAPlugin__factory,
+  SafeECDSARecoveryPlugin,
+  Safe__factory,
+  SafeECDSAFactory,
+} from '../../../../demos/inpage/hardhat/typechain-types';
+import EthereumRpc from '../EthereumRpc';
+import IAccount from './IAccount';
+import receiptOf from '../helpers/receiptOf';
+import { executeContractCallWithSigners } from './execution';
+import assert from '../helpers/assert';
+import getContracts from './getContracts';
+
+export const SafeECDSAAccountData = z.object({
+  type: z.literal('SafeECDSAAccount'),
+  address: z.string(),
+  privateKey: z.string(),
+  ownerAddress: z.string(),
+  recoveryAddress: z.optional(z.string()),
+});
+
+export type SafeECDSAAccountData = z.infer<typeof SafeECDSAAccountData>;
+
+// Cost of validating a signature or whatever verification method is in place.
+const baseVerificationGas = 100_000n;
+
+export default class SafeECDSAAccountWrapper implements IAccount {
+  type = 'SafeECDSAAccount';
+
+  constructor(
+    public address: string,
+    public privateKey: string,
+    public ownerAddress: string,
+    public recoveryAddress: string | undefined,
+    public provider: ethers.BrowserProvider,
+  ) {}
+
+  static fromData(data: SafeECDSAAccountData, provider: ethers.BrowserProvider) {
+    return new SafeECDSAAccountWrapper(
+      data.address,
+      data.privateKey,
+      data.ownerAddress,
+      data.recoveryAddress,
+      provider,
+    );
+  }
+
+  toData(): SafeECDSAAccountData {
+    return {
+      type: 'SafeECDSAAccount',
+      address: this.address,
+      privateKey: this.privateKey,
+      ownerAddress: this.ownerAddress,
+      recoveryAddress: this.recoveryAddress,
+    };
+  }
+
+  static async createRandom(
+    // TODO: clean up parameters needed for getContracts
+    provider: ethers.BrowserProvider,
+    admin: ethers.Signer
```

Comment:
Added these TODOs because I wasn't sure if we wanted to have to pass an admin signer to this function.  However, it's needed right now in case we need to deploy the contracts in the `getContracts()` function